src/xpython_interpreter.cpp

```diff
@@ -159,7 +136,35 @@ namespace xpyt
         return result;
     }
 
-    void python_interpreter::input_reply_impl(const std::string& /*value*/)
+    void xpython_interpreter::input_reply_impl(const std::string& /*value*/)
+    {
+    }
+
+    void xpython_interpreter::redirect_output()
     {
+        // In Python:
+        // import sys and import xeus_python_logger
+        py::module sys = py::module::import("sys");
+        py::module xeus_python_logger = py::module::import("xeus_python_logger");
+
+        // Create XPythonLogger instance
+        py::object out_logger = xeus_python_logger.attr("XPythonLogger")();
+        py::object err_logger = xeus_python_logger.attr("XPythonLogger")();
+
+        py::cpp_function publish_stdout_stream = [this](const std::string& message){
+            this->publish_stream("stdout", message);
+        };
+
+        py::cpp_function publish_stderr_stream = [this](const std::string& message){
+            this->publish_stream("stderr", message);
+        };
+
+        // Add publish_stream as a logger function
+        out_logger.attr("add_logger")(publish_stdout_stream);
+        err_logger.attr("add_logger")(publish_stderr_stream);
+
+        // And replace sys.stdout by the XPythonLogger instance
+        sys.attr("stdout") = out_logger;
+        sys.attr("stderr") = err_logger;
```

Comment:
You may want to save the original ones in `__std*__`

Reply:
Good point, thanks!